src/components/Select/Select.tsx

```diff
@@ -141,6 +141,10 @@ const Select: React.FunctionComponent<React.PropsWithChildren<SelectProps>> = ({
     }
   }, [])
 
+  useEffect(() => {
+    setSelectedOptionIndex(defaultOptionIndex)
+  }, [defaultOptionIndex])
+
```

Comment:
not a fan of using `defaultOptionIndex` as `value`. but it's a flawed design on this Select component.
I will let this pass this time